code/modules/clothing/head/misc_special.dm

```diff
@@ -184,6 +184,26 @@
 /obj/item/clothing/head/kitty/cursed
 	canremove = 0
 
+/obj/item/clothing/head/kitty/equipped(mob/M, var/slot)
+	..()
+	var/mob/living/carbon/human/H = M
+	if(!istype(H))
+		return
+	if(H.head == src)
```

Comment:
Compare the `slot` variable instead of checking the mob's inventory variables directly.